pkg/operator/controllers/dnsmasq/dnsmasq.go

```diff
@@ -88,51 +89,55 @@ func ignition2Config(clusterDomain, apiIntIP, ingressIP string, gatewayDomains [
 		return nil, err
 	}
 
-	ign := &ign2types.Config{
-		Ignition: ign2types.Ignition{
-			Version: ign2types.MaxVersion.String(),
+	ign := &ign3types.Config{
+		Ignition: ign3types.Ignition{
+			// This Ignition Config version should be kept up to date with the default
+			// rendered Ignition Config version from the Machine Config Operator version
+			// on the lowest OCP version we support (4.7).
+			Version: semver.Version{
+				Major: 3,
+				Minor: 2,
+			}.String(),
 		},
-		Storage: ign2types.Storage{
-			Files: []ign2types.File{
+		Storage: ign3types.Storage{
+			Files: []ign3types.File{
 				{
-					Node: ign2types.Node{
-						Filesystem: "root",
-						Overwrite:  to.BoolPtr(true),
-						Path:       "/etc/" + configFileName,
-						User: &ign2types.NodeUser{
-							Name: "root",
+					Node: ign3types.Node{
+						Overwrite: ignutil.BoolToPtr(true),
+						Path:      "/etc/" + configFileName,
+						User: ign3types.NodeUser{
+							Name: ignutil.StrToPtr("root"),
 						},
 					},
-					FileEmbedded1: ign2types.FileEmbedded1{
-						Contents: ign2types.FileContents{
-							Source: dataurl.EncodeBytes(config),
+					FileEmbedded1: ign3types.FileEmbedded1{
+						Contents: ign3types.Resource{
+							Source: ignutil.StrToPtr(dataurl.EncodeBytes(config)),
 						},
-						Mode: to.IntPtr(0644),
+						Mode: ignutil.IntToPtr(0644),
 					},
 				},
 				{
-					Node: ign2types.Node{
-						Filesystem: "root",
-						Overwrite:  to.BoolPtr(true),
-						Path:       "/usr/local/bin/" + prescriptFileName,
-						User: &ign2types.NodeUser{
-							Name: "root",
+					Node: ign3types.Node{
+						Overwrite: ignutil.BoolToPtr(true),
+						Path:      "/usr/local/bin/" + prescriptFileName,
+						User: ign3types.NodeUser{
+							Name: ignutil.StrToPtr("root"),
 						},
 					},
-					FileEmbedded1: ign2types.FileEmbedded1{
-						Contents: ign2types.FileContents{
-							Source: dataurl.EncodeBytes(startpre),
+					FileEmbedded1: ign3types.FileEmbedded1{
+						Contents: ign3types.Resource{
+							Source: ignutil.StrToPtr(dataurl.EncodeBytes(startpre)),
 						},
-						Mode: to.IntPtr(0744),
+						Mode: ignutil.IntToPtr(0744),
```

Comment:
I believe there was an effort to remove our dependency on ignition util not too long ago. Any reason to use it over `to`?

Reply:
This was just the result of me carrying over the existing ignition3 config from installer-aro, which used these conversion functions: https://github.com/jewzaam/installer-aro/blob/96c4ea6a1ddb8d3885bae9fe7648462a9d6580d1/pkg/aro/dnsmasq/dnsmasq.go#L275 

I'm happy to switch these back over.

Reply:
Made the change